File: martial-world/frontend/patch.js
const fs = require('fs');

const files = ['karate.html', 'kungfu.html', 'taekwondo.html', 'muaythai.html', 'mma.html', 'boxing.html', 'kickboxing.html'];

for (const f of files) {
    let content = fs.readFileSync(f, 'utf-8');

    // Step 1: Add currentUser to updateSyllabus
    if (!content.includes("const currentUser = localStorage.getItem('currentUser') || 'guest';")) {
        content = content.replace(
            /function updateSyllabus\(\) \{/,
            "function updateSyllabus() {\n      const currentUser = localStorage.getItem('currentUser') || 'guest';"
        );
    }

    // Step 2: Update checkboxId
    content = content.replace(
        /const checkboxId = `\$\{belt\}-\$\{monthIndex\}-day\$\{day\}-goal\$\{goalIndex\}`;/g,
        "const checkboxId = `${currentUser}-${belt}-${monthIndex}-day${day}-goal${goalIndex}`;"
    );

    // Step 3: Update noteId
    content = content.replace(
        /const noteId = `\$\{belt\}-\$\{monthIndex\}-day\$\{day\}-note`;/g,
        "const noteId = `${currentUser}-${belt}-${monthIndex}-day${day}-note`;"
    );

    fs.writeFileSync(f, content, 'utf-8');
}

console.log("Patching complete.");

File: martial-world/frontend/patch.py
import os
import glob
import re

files = ['karate.html', 'kungfu.html', 'taekwondo.html', 'muaythai.html', 'mma.html', 'boxing.html', 'kickboxing.html']

for f in files:
    with open(f, 'r', encoding='utf-8') as file:
        content = file.read()

    # Step 1: Add currentUser to updateSyllabus
    if "const currentUser = localStorage.getItem('currentUser') || 'guest';" not in content:
        content = re.sub(
            r'function updateSyllabus\(\) \{',
            r"function updateSyllabus() {\n      const currentUser = localStorage.getItem('currentUser') || 'guest';",
            content
        )
    
    # Step 2: Update checkboxId
    content = re.sub(
        r'const checkboxId = `\$\{belt\}-\$\{monthIndex\}-day\$\{day\}-goal\$\{goalIndex\}`;',
        r'const checkboxId = `${currentUser}-${belt}-${monthIndex}-day${day}-goal${goalIndex}`;',
        content
    )
    
    # Step 3: Update noteId
    content = re.sub(
        r'const noteId = `\$\{belt\}-\$\{monthIndex\}-day\$\{day\}-note`;',
        r'const noteId = `${currentUser}-${belt}-${monthIndex}-day${day}-note`;',
        content
    )

    with open(f, 'w', encoding='utf-8') as file:
        file.write(content)

print("Patching complete.")
